test/e2e/subscription_e2e_test.go

```diff
@@ -62,6 +64,82 @@ var _ = Describe("Subscription", func() {
 		TeardownNamespace(generatedNamespace.GetName())
 	})
 
+	When("a registry server is not started for a grpc-type CatalogSource", func() {
+		var (
+			catsrc *v1alpha1.CatalogSource
+		)
+
+		BeforeEach(func() {
+			catsrc = &v1alpha1.CatalogSource{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace:    generatedNamespace.GetName(),
+					GenerateName: "without-registry-server-",
+				},
+				Spec: v1alpha1.CatalogSourceSpec{
+					SourceType: v1alpha1.SourceTypeGrpc,
+					Image:      "@", // bad image ref, pod creation will fail
+				},
+			}
+			Expect(ctx.Ctx().Client().Create(context.Background(), catsrc)).To(Succeed())
+		})
+
+		AfterEach(func() {
+			Eventually(func() error {
+				if catsrc == nil {
+					return nil
+				}
+				return ctx.Ctx().Client().Delete(context.Background(), catsrc)
+			}).Should(Or(
+				Succeed(),
+				WithTransform(errors.IsNotFound, BeTrue()),
+			))
+		})
+
+		It("should indicate ErrorPreventedResolution on a dependent Subscription status", func() {
+			sub := v1alpha1.Subscription{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace:    generatedNamespace.GetName(),
+					GenerateName: "test-subscription",
+				},
+				Spec: &v1alpha1.SubscriptionSpec{
+					Package: "whatever",
+				},
+			}
+			Expect(ctx.Ctx().Client().Create(context.Background(), &sub)).To(Succeed())
+
+			getCondition := func() (v1alpha1.SubscriptionCondition, error) {
+				if err := ctx.Ctx().Client().Get(context.Background(), client.ObjectKeyFromObject(&sub), &sub); err != nil {
+					return v1alpha1.SubscriptionCondition{}, err
+				}
+				cond := sub.Status.GetCondition(v1alpha1.SubscriptionResolutionFailed)
+				ctx.Ctx().Logf("%#v\n", cond)
+				return v1alpha1.SubscriptionCondition{
+					Type:   cond.Type,
+					Reason: cond.Reason,
+					Status: cond.Status,
+				}, nil
+			}
+
+			// this doesn't seem very robust. basically, this subscription condition should arrive directly at True/ErrorPreventedResolution without passing through any other non-Unknown states
```

Comment:
Not happy with this test, but it consistently reproduced the bug for me on master.